lib/portage/module.py

```diff
@@ -107,19 +110,27 @@ def _get_all_modules(self):
         @rtype: dictionary of module_plugins
         """
         module_dir = self._module_path
-        importables = []
         names = os.listdir(module_dir)
-        for entry in names:
-            # skip any __init__ or __pycache__ files or directories
-            if entry.startswith("__"):
-                continue
+
+        def _a_real_module(entry):
             try:
                 # test for statinfo to ensure it should a real module
                 # it will bail if it errors
                 os.lstat(os.path.join(module_dir, entry, "__init__.py"))
-                importables.append(entry)
             except EnvironmentError:
-                pass
+                return False
+            return True
+
+        # The importables list cannot be a generator.
```

Comment:
💯

Reply:
```python
>>> the_list = [i for i in range(10)]
>>> the_generator = (i for i in range(10))
>>> list(the_list)
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
>>> list(the_list)
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
>>> list(the_generator)
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
>>> list(the_generator)
[]
```

Reply:
Yep! (I was originally just saying I liked the comment, but thanks for elaborating -- actually seeing it is helpful!)